packages/app/src/elements/history-entry-dialog.ts

```diff
@@ -0,0 +1,121 @@
+import { translate as $l } from "@padloc/locale/src/translate";
+import { ItemHistory, Field } from "@padloc/core/src/item";
+import { confirm } from "../lib/dialog";
+import { Dialog } from "./dialog";
+import { html, css } from "lit";
+import { customElement } from "lit/decorators.js";
+import { repeat } from "lit/directives/repeat.js";
+
+@customElement("pl-history-entry-dialog")
+export class HistoryEntryDialog extends Dialog<ItemHistory, boolean> {
+    private _historyEntry: ItemHistory;
+
+    static styles = [
+        ...Dialog.styles,
+        css`
+            :host {
+                --pl-dialog-max-width: 40em;
+            }
+        `,
+    ];
+
+    // TODO: show vault changes
+
+    renderContent() {
+        if (!this._historyEntry) {
+            return html``;
+        }
+
+        const { fieldsChanged, date } = this._historyEntry;
+
+        return html`
+            <div class="padded vertical spacing layout">
+                <h1 class="big margined text-centering">${$l("History Entry")}</h1>
+                <h2 class="margined text-centering">${$l("Created at {0}", date as unknown as string)}</h2>
+
+                <div class="stretch"></div>
+
+                ${fieldsChanged.includes("name") ? this._showChangedNameContent() : null}
+                ${fieldsChanged.includes("tags") ? this._showChangedTagsContent() : null}
+                ${fieldsChanged.includes("fields") ? this._showChangedFieldsContent() : null}
+
+                <div class="horizontal evenly stretching spacing layout top-margined">
+                    <pl-button class="primary" @click=${() => this._restore()}>
+                        ${$l("Restore item to this version")}
+                    </pl-button>
+                    <pl-button @click=${this.dismiss}> ${$l("Cancel")} </pl-button>
+                </div>
+            </div>
+        `;
```

Comment:
I would just display the entire item as it was at the time. In general in would view history entries as a snapshot of the entire item, rather than a change set. I think it's much easier for users to digest that way.

Reply:
I think that's harder to grasp at what changed, though. If you have an item with 10 fields, and only the tags or name changed, it'll be harder to understand that if you're seeing everything.

I wish we could just display the fields that changed, but that's harder to do.

Reply:
I think you are focusing too much on displaying changes. Trust me, people care much less about what has changed and much more about simply restoring an item to a certain point in time. In fact, I would probably not display changes at all but if you think it would make for a better UX we can highlight the changed properties somehow.